src/platform/silabs/SiWx917/wifi/wfx_notify.cpp

```diff
@@ -199,7 +218,22 @@ void wfx_retry_interval_handler(bool is_wifi_disconnection_event, uint16_t retry
         if (retryJoin < MAX_JOIN_RETRIES_COUNT)
         {
             SILABS_LOG("wfx_retry_interval_handler : Next attempt after %d Seconds", CONVERT_MS_TO_SEC(WLAN_RETRY_TIMER_MS));
+#if SL_ICD_ENABLED
+            // TODO: cleanup the retry logic MATTER-1921
+            if (!BaseApplication::sAppDelegate.isComissioningStarted)
```

Comment:
This makes me realize that `isComissioningStarted` should be private and if needed outside of the BaseApplication class a getter function can be created.
We wouldn't want `isComissioningStarted`t be changed outside of the delegate implementation.

Reply:
Made this private